contracts/Vault.vy

```diff
@@ -853,13 +852,17 @@ def _issueSharesForAmount(to: address, amount: uint256) -> uint256:
         # Mint amount of shares based on what the Vault is managing overall
         # NOTE: if sqrt(token.totalSupply()) > 1e39, this could potentially revert
         shares =  amount * totalSupply / self._freeFunds()  # dev: no free funds
+        self.totalSupply = totalSupply + shares
     else:
-        # No existing shares, so mint 1:1
-        shares = amount
+        # Permanently lock the first MINIMUM_LIQUIDITY tokens
+        toBurn: uint256 = MINIMUM_LIQUIDITY
+        shares = amount - toBurn
+        self.totalSupply = totalSupply + amount
+        self.balanceOf[ZERO_ADDRESS] += toBurn
+        log Transfer(ZERO_ADDRESS, ZERO_ADDRESS, toBurn)
```

Comment:
this would confuse indexers as they won't be able to reconcile the total supply, mistaking this event for a burn.

maybe an easier approach could be to just not allow deposits/withdrawals which result in a total supply in a range of `0 < x < dust`?